spring-cloud-gcp-autoconfigure/pom.xml

```diff
@@ -50,11 +50,9 @@
 
         <!-- Cloud SQL -->
 
-        <!-- HikariCP used as default DataSource -->
         <dependency>
-            <groupId>com.zaxxer</groupId>
-            <artifactId>HikariCP</artifactId>
-            <scope>compile</scope>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-jdbc</artifactId>
```

Comment:
It seems odd to have a dependency on a starter in the autoconfigure module, albeit an optional one.
I don't see any examples of this in the Spring Boot project, and for starters in this project we can't do it because it would create a circular dependency.

Reply:
Well, agree. That really was a speculation on my side.
But I did it because I see how you use here already `spring-cloud-starter-sleuth`.

OTOH I don't think that it is a big problem since the `starter` brings for us just dependencies so we don't need to copy/paste them.

Not sure what you mean about "circular dependency" because this is `optional` here, plus `spring-boot-starter-jdbc` doesn't use our `spring-cloud-gcp-autoconfigure`. 

But again: since it is `optional` here and we definitely copy/paste some dependencies to the SQL starter, that sounds like it is much better to use this `spring-boot-starter-jdbc` as a strong dependency from the starters. But again: this is a story of the issue #371 .

If you feel strong, we can reconsider this line when we do that issue.

Reply:
OK, sgtm.